cpp/src/arrow/csv/reader_test.cc

```diff
@@ -216,5 +216,42 @@ TEST(StreamingReaderTests, NestedParallelism) {
   TestNestedParallelism(thread_pool, table_factory);
 }
 
+TEST(ReaderOptionsTests, SkipRowsAfterNames) {
```

Comment:
Can you add a test case where `autogenerate_column_names` is set to `true` (i.e. there is no header in the CSV file)?

Reply:
Done